projects/dsp-ui/src/lib/search/fulltext-search/fulltext-search.component.spec.ts

```diff
@@ -0,0 +1,399 @@
+        expect(searchInput).toBeDefined();
+        searchInput.click();
+        testHostFixture.detectChanges();
+
+        // click on the Clear List button to erase the search list
+        const clearListBtn = fulltextSearchComponentDe.query(By.css('button.clear-list-btn')).nativeElement;
+        expect(clearListBtn).toBeDefined();
+        clearListBtn.click();
+        testHostFixture.detectChanges();
+
+        expect(localStorage.getItem('prevSearch')).toBe(null);
+
+    });
+
+    it('should clear the search list - solution 2', () => {
+        const prevSearchArray: PrevSearchItem[] = [
+            { projectIri: 'http://rdfh.ch/projects/0803', projectLabel: 'incunabula', query: 'one thing' },
+            { projectIri: 'http://rdfh.ch/projects/0803', projectLabel: 'incunabula', query: 'two things' },
+            { projectIri: 'http://rdfh.ch/projects/0801', projectLabel: 'anything', query: 'hello world' }
+        ];
+
+        localStorage.setItem('prevSearch', JSON.stringify(prevSearchArray));
+        expect(localStorage.getItem('prevSearch')).toBeDefined();
+
+        testHostComponent.fulltextSearch.resetPrevSearch();
+
+        expect(localStorage.getItem('prevSearch')).toBe(null);
+    });
+
+    it('should remove one item of the search list - solution 1', () => {
+        const prevSearchArray: PrevSearchItem[] = [
+            { projectIri: 'http://rdfh.ch/projects/0803', projectLabel: 'incunabula', query: 'one thing' },
+            { projectIri: 'http://rdfh.ch/projects/0803', projectLabel: 'incunabula', query: 'two things' },
+            { projectIri: 'http://rdfh.ch/projects/0801', projectLabel: 'anything', query: 'hello world' }
+        ];
+
+        localStorage.setItem('prevSearch', JSON.stringify(prevSearchArray));
+        expect(localStorage.getItem('prevSearch')).toBeDefined();
+
+        // click in the search input to open the search panel
+        const searchInput = fulltextSearchComponentDe.query(By.css('input.kui-fulltext-search-input')).nativeElement;
+        expect(searchInput).toBeDefined();
+        searchInput.click();
+        testHostFixture.detectChanges();
+
+        // click on the close icon to remove the item of the list
+        const closeItemBtn = fulltextSearchComponentDe.query(By.css('mat-icon.mat-list-close-icon')).nativeElement;
+        expect(closeItemBtn).toBeDefined();
+        closeItemBtn.click();
+        testHostFixture.detectChanges();
+
+        const newPrevSearchArray: PrevSearchItem[] = [
+            { projectIri: 'http://rdfh.ch/projects/0803', projectLabel: 'incunabula', query: 'two things' },
+            { projectIri: 'http://rdfh.ch/projects/0803', projectLabel: 'incunabula', query: 'one thing' }
+        ];
+        expect(localStorage.getItem('prevSearch')).toEqual(JSON.stringify(newPrevSearchArray));
+    });
+
+    // todo: fix test - ERROR: Cannot read property 'indexOf' of null
```

Comment:
@tobiasschweizer I get the following error: Cannot read property 'indexOf' of null. I don't understand. This test calls directly the method I want to check (l.339). Any idea?

Reply:
`prevSearch: PrevSearchItem[] = JSON.parse(localStorage.getItem('prevSearch'));` is set to `null` in case the item does not exist. I think you should handle the initialisation of  prevSearch differently and set it to an empty array if there is no previous search.

Reply:
I do not understand why here `prevSearch` is set to null. The test should remove one item of the search list so I need a list. The solution 1 works.

Reply:
Independent of you test, `prevSearch: PrevSearchItem[] = JSON.parse(localStorage.getItem('prevSearch'));` is not good because the member `prevSearch` could be assigned `null`.

Rather you should do it in `ngOnInit`:

```

const prevSearchOption = JSON.parse(localStorage.getItem('prevSearch'));
if (prevSearchOption !== null) {
   this.prevSearch = prevSearchOption;
} else {
   this.prevSearch = [];
}
```

Like this, you always have an array which is compatible with `prevSearch: PrevSearchItem[]`.

We should think about using `strictNullChecks` (https://www.typescriptlang.org/docs/handbook/compiler-options.html), so this is enforced at compile time.

Reply:
prevSearch is always set to [] even if the localStorage has a key `prevSearch`

Reply:
But wouldn't it be set to `null` if there is no item fot he given key in the localStorage? `localStorage.getItem` returns `null` if the key does not exist, right?

My understanding is that this lead to ` Cannot read property 'indexOf' of null`

Reply:
`prevSearch` is undefined and `prevSearchOption` is null despite the fact I set the localStorage in the beforeEach... Is there another setting to do?

Reply:
I think I would have to try to go through the code myself. Maybe then it gets clearer for me.